### PR TITLE
Local Volume CleanUp Added

### PR DESCRIPTION
**Pull request by ashimagarg10**

### CODE REVIEW

File: controllers/cleanupoperator_controller.go [hunk 1]
```diff
@@ -174,8 +193,40 @@ func (r *CleanUpOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Requ
...
+
+				localVolumeOpFound := true
+				localVolumeOperator := &appsv1.Deployment{}
+				err = r.Get(ctx, types.NamespacedName{Name: localVolumeOperatorName, Namespace: namespace}, localVolumeOperator)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						log.Info("Local Volume Operator not found. Ignoring...", "name", localVolumeOperatorName)
+						localVolumeOpFound = false
+					}
+					log.Error(err, "Failed to get Local Volume Operator", "name", localVolumeOperatorName)
```
Comment (mssachan) [inline controllers/cleanupoperator_controller.go]: Put this in `else` part.
Please update same for `netapp`. This is a bug in code.

Comment (mssachan) [reply controllers/cleanupoperator_controller.go]: else {
  log.Error(err, "Failed to get Local Volume Operator", "name", localVolumeOperatorName)
  return ctrl.Result{}, err
}

File: controllers/localVolumeCleanUp.go [hunk 2]
```diff
@@ -1,76 +1,124 @@
...
+func (cr *CleanUpOperatorReconciler) localVolumeCleanUp(ctx context.Context, namespace string) error {
+	localVolumeStatus := true
+	localDisk := &localv1.LocalVolume{}
+	err := cr.Get(ctx, types.NamespacedName{Name: "local-disk", Namespace: namespace}, localDisk)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("LocalVolume 'local-disk' not found")
+			localVolumeStatus = false
+		}
+		fmt.Println(err, "Error in getting LocalVolume 'local-disk'")
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: Put this in `else` part.

File: controllers/localVolumeCleanUp.go [hunk 3]
```diff
@@ -1,76 +1,124 @@
...
-// 	fmt.Println("Mounted Paths Removed....")
-// 	return true
-// }
+	// Find PVs
+	pvList := &corev1.PersistentVolumeList{}
+	err = cr.List(ctx, pvList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			fmt.Println("PVs not found")
+			return err
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: why are we returning error if there is no PVs?
Also does List return NotFound error when there is no PV? It is not returning empty list?

File: controllers/localVolumeCleanUp.go [hunk 4]
```diff
@@ -1,76 +1,124 @@
...
+	// PV Deletion
+	for _, pv := range pvList.Items {
+		if strings.HasPrefix(pv.Name, "local-pv-") {
+			fmt.Println("PV status- ", pv.Status.Phase)
+			// if pv.Status.Phase == "Available" {
+			err = cr.Delete(ctx, &pv)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					fmt.Println("PV not found")
+					return err
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: Same here

File: controllers/cleanupoperator_controller.go (hunk 1 above)
Comment (ashimagarg10) [reply controllers/cleanupoperator_controller.go]: Done

File: controllers/localVolumeCleanUp.go (hunk 2 above)
Comment (ashimagarg10) [reply controllers/localVolumeCleanUp.go]: Done

File: controllers/localVolumeCleanUp.go (hunk 3 above)
Comment (ashimagarg10) [reply controllers/localVolumeCleanUp.go]: Right, no need to return the error.
But: As per this  https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/client#Reader
List returns an error on failure.
Please once confirm.

**Comment by ashimagarg10** [comment]
Logs : (Steps followed)
1. First Deploy Operator
2. Install Local Volume Template
3. Install Trident Template
4. Delete Local Volume Assignment
5. Delete Trident Assignment

```
% oc logs -f controller-manager-8556f857cc-j6q26 -n system -c manager
I0712 10:43:58.047970       1 request.go:655] Throttling request took 1.026497824s, request: GET:https://172.21.0.1:443/apis/network.operator.openshift.io/v1?timeout=32s
2021-07-12T10:43:59.499Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
2021-07-12T10:43:59.500Z	INFO	setup	starting manager
I0712 10:43:59.500515       1 leaderelection.go:243] attempting to acquire leader lease system/e9d90034.ibm.com...
2021-07-12T10:43:59.500Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I0712 10:43:59.531659       1 leaderelection.go:253] successfully acquired lease system/e9d90034.ibm.com
2021-07-12T10:43:59.531Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"system","name":"e9d90034.ibm.com","uid":"4fd582ba-988d-45f0-9ceb-5af5d8d4fa03","apiVersion":"v1","resourceVersion":"18075549"}, "reason": "LeaderElection", "message": "controller-manager-8556f857cc-j6q26_f3c8f152-8bcc-4841-b1fc-f04da42253b8 became leader"}
2021-07-12T10:43:59.531Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"Lease","namespace":"system","name":"e9d90034.ibm.com","uid":"ecd5e4e0-dd98-4cac-aaa8-170a6d9749e7","apiVersion":"coordination.k8s.io/v1","resourceVersion":"18075550"}, "reason": "LeaderElection", "message": "controller-manager-8556f857cc-j6q26_f3c8f152-8bcc-4841-b1fc-f04da42253b8 became leader"}
2021-07-12T10:43:59.532Z	INFO	controller-runtime.manager.controller.cleanupoperator	Starting EventSource	{"reconciler group": "cleanup.ibm.com", "reconciler kind": "CleanUpOperator", "source": "kind source: /, Kind="}
2021-07-12T10:43:59.632Z	INFO	controller-runtime.manager.controller.cleanupoperator	Starting Controller	{"reconciler group": "cleanup.ibm.com", "reconciler kind": "CleanUpOperator"}
2021-07-12T10:43:59.633Z	INFO	controller-runtime.manager.controller.cleanupoperator	Starting workers	{"reconciler group": "cleanup.ibm.com", "reconciler kind": "CleanUpOperator", "worker count": 1}
2021-07-12T10:45:13.596Z	INFO	controllers.CleanUpOperator	custom finalizer added to CleanupOperator	{"cleanupoperator": "system/cleanupoperator-localvolume"}
2021-07-12T10:45:13.698Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:13.698Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:13.704Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:13.724Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:13.766Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:13.847Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:14.007Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:14.328Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:14.968Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:16.249Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:18.810Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:23.930Z	INFO	controllers.CleanUpOperator	Local Volume Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:45:34.186Z	INFO	controllers.CleanUpOperator	custom finalizer added to Local Volume Operator	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:46:50.771Z	INFO	controllers.CleanUpOperator	custom finalizer added to CleanupOperator	{"cleanupoperator": "system/cleanupoperator-netapp"}
2021-07-12T10:46:50.771Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:50.771Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:50.777Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:50.797Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:50.838Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:50.918Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:51.079Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:51.400Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:52.040Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:53.321Z	INFO	controllers.CleanUpOperator	Trident Operator not found. Check again...	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:46:55.910Z	INFO	controllers.CleanUpOperator	custom finalizer added to Trident Operator	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
Local Volume
I0712 10:49:19.265079       1 request.go:655] Throttling request took 1.037856507s, request: GET:https://172.21.0.1:443/apis/machineconfiguration.openshift.io/v1?timeout=32s
LocalVolume 'local-disk' not found
PV status-  Available
PV(s) Deleted.....
in ExecuteCommand

in ExecuteCommand

in ExecuteCommand

Mounted Paths Removed....
localvolumediscoveries.local.storage.openshift.io
localvolumediscoveryresults.local.storage.openshift.io
localvolumes.local.storage.openshift.io
localvolumesets.local.storage.openshift.io
2021-07-12T10:49:33.321Z	INFO	controllers.CleanUpOperator	custom finalizer removed from local volume Operator	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-12T10:49:33.321Z	INFO	controllers.CleanUpOperator	Local Volume Template Cleaned Successfully!!!	{"cleanupoperator": "system/cleanupoperator-localvolume"}
2021-07-12T10:49:33.504Z	INFO	controllers.CleanUpOperator	custom finalizer removed from CleanupOperator	{"cleanupoperator": "system/cleanupoperator-localvolume"}
2021-07-12T10:49:33.504Z	INFO	controllers.CleanUpOperator	CleanUpOperator resource not found. Ignoring since object must be deleted	{"cleanupoperator": "system/cleanupoperator-localvolume"}
E0712 10:49:34.288557       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
E0712 10:49:35.346551       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: failed to list *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
E0712 10:49:37.700209       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: failed to list *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
E0712 10:49:41.810551       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: failed to list *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
E0712 10:49:50.104893       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: failed to list *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
E0712 10:50:11.601905       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.LocalVolume: failed to list *v1.LocalVolume: the server could not find the requested resource (get localvolumes.local.storage.openshift.io)
NetApp Trident
2021-07-12T10:50:21.257Z	INFO	controllers.CleanUpOperator	removed finalizers from CR	{"name": "169.62.34.218"}
2021-07-12T10:50:21.290Z	INFO	controllers.CleanUpOperator	removed finalizers from CR	{"name": "169.62.34.219"}
2021-07-12T10:50:21.305Z	INFO	controllers.CleanUpOperator	removed finalizers from CR	{"name": "169.62.34.222"}
2021-07-12T10:50:21.561Z	INFO	controllers.CleanUpOperator	removed finalizers from CR	{"name": "trident"}
2021-07-12T10:50:21.604Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentbackends.trident.netapp.io"}
2021-07-12T10:50:21.690Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentbackends.trident.netapp.io"}
2021-07-12T10:50:21.814Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentsnapshots.trident.netapp.io"}
2021-07-12T10:50:21.948Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentsnapshots.trident.netapp.io"}
2021-07-12T10:50:22.050Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentstorageclasses.trident.netapp.io"}
2021-07-12T10:50:22.166Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentstorageclasses.trident.netapp.io"}
2021-07-12T10:50:22.215Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridenttransactions.trident.netapp.io"}
2021-07-12T10:50:22.253Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridenttransactions.trident.netapp.io"}
2021-07-12T10:50:22.271Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentvolumes.trident.netapp.io"}
2021-07-12T10:50:22.293Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentvolumes.trident.netapp.io"}
2021-07-12T10:50:22.319Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentversions.trident.netapp.io"}
2021-07-12T10:50:22.342Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentversions.trident.netapp.io"}
2021-07-12T10:50:22.360Z	INFO	controllers.CleanUpOperator	removed finalizers from CRD	{"name": "tridentnodes.trident.netapp.io"}
2021-07-12T10:50:22.409Z	INFO	controllers.CleanUpOperator	CRD deleted	{"name": "tridentnodes.trident.netapp.io"}
2021-07-12T10:50:22.409Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "removeCRDs", "time": 0.847381877}
2021-07-12T10:50:22.451Z	INFO	controllers.CleanUpOperator	custom finalizer removed from Trident Operator	{"cleanupoperator": "system/cleanupoperator-netapp", "name": "trident-operator"}
2021-07-12T10:50:22.452Z	INFO	controllers.CleanUpOperator	NetApp Trident Template Cleaned Successfully!!!	{"cleanupoperator": "system/cleanupoperator-netapp"}
2021-07-12T10:50:22.610Z	INFO	controllers.CleanUpOperator	custom finalizer removed from CleanupOperator	{"cleanupoperator": "system/cleanupoperator-netapp"}
2021-07-12T10:50:22.610Z	INFO	controllers.CleanUpOperator	CleanUpOperator resource not found. Ignoring since object must be deleted	{"cleanupoperator": "system/cleanupoperator-netapp"}
E0712 10:50:23.425742       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentVersion: the server could not find the requested resource (get tridentversions.trident.netapp.io)
E0712 10:50:23.611717       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentNode: the server could not find the requested resource (get tridentnodes.trident.netapp.io)
E0712 10:50:24.395528       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentVersion: failed to list *v1.TridentVersion: the server could not find the requested resource (get tridentversions.trident.netapp.io)
E0712 10:50:24.974904       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentNode: failed to list *v1.TridentNode: the server could not find the requested resource (get tridentnodes.trident.netapp.io)
E0712 10:50:26.625156       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentNode: failed to list *v1.TridentNode: the server could not find the requested resource (get tridentnodes.trident.netapp.io)
E0712 10:50:26.837410       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.6/tools/cache/reflector.go:167: Failed to watch *v1.TridentVersion: failed to list *v1.TridentVersion: the server could not find the requested resource (get tridentversions.trident.netapp.io)
```

File: controllers/localVolumeCleanUp.go [hunk 5]
```diff
@@ -1,76 +1,119 @@
...
-// 	fmt.Println("Mounted Paths Removed....")
-// 	return true
-// }
+	// Find PVs
+	pvList := &corev1.PersistentVolumeList{}
+	pvfound := true
+	err = cr.List(ctx, pvList)
+	if err != nil {
+		fmt.Println(err, "Error in getting PVs")
+		pvfound = false
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: I think we need to return err if there is any error while getting PV list.
I asked just not to remove IsNotFound error

File: controllers/localVolumeCleanUp.go [hunk 6]
```diff
@@ -1,76 +1,119 @@
...
+		fmt.Println("PV(s) Deleted.....")
+	}
+
+	// Remove Mounted Path
+	nodesList := &corev1.NodeList{}
+	nodesfound := true
+	err = cr.List(ctx, nodesList)
+	if err != nil {
+		fmt.Println(err, "Error in getting Nodes List")
+		nodesfound = true
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: Similarly here.
I think we need to return err if there is any error while getting PV list.
I asked just not to remove IsNotFound error

Comment (ashimagarg10) [reply controllers/localVolumeCleanUp.go]: Done

**Comment by ashimagarg10** [comment]
Logs after adding time:

```
Local Volume
I0713 11:02:54.679293       1 request.go:655] Throttling request took 1.03834788s, request: GET:https://172.21.0.1:443/apis/samples.operator.openshift.io/v1?timeout=32s
LocalVolume 'local-disk' not found
PV(s) Deleted.....
in ExecuteCommand

in ExecuteCommand

in ExecuteCommand

Mounted Paths Removed....
2021-07-13T11:03:39.148Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "Mounted Path Removal", "time": 12.162364861}
2021-07-13T11:03:39.148Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "Mounted Path Removal", "time": 26.573681598}
2021-07-13T11:03:39.148Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "Mounted Path Removal", "time": 42.553525479}
2021-07-13T11:03:39.148Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "localVolumeCleanUp", "time": 45.519454942}
localvolumediscoveries.local.storage.openshift.io
localvolumediscoveryresults.local.storage.openshift.io
localvolumes.local.storage.openshift.io
localvolumesets.local.storage.openshift.io
2021-07-13T11:03:41.099Z	INFO	controllers.CleanUpOperator	Time to complete	{"method": "RemoveCRDs", "time": 1.951649909}
2021-07-13T11:03:41.145Z	INFO	controllers.CleanUpOperator	custom finalizer removed from local volume Operator	{"cleanupoperator": "system/cleanupoperator-localvolume", "name": "local-storage-operator"}
2021-07-13T11:03:41.146Z	INFO	controllers.CleanUpOperator	Local Volume Template Cleaned Successfully!!!	{"cleanupoperator": "system/cleanupoperator-localvolume"}
2021-07-13T11:03:41.179Z	INFO	controllers.CleanUpOperator	custom finalizer removed from CleanupOperator	{"cleanupoperator": "system/cleanupoperator-localvolume"}

```

File: controllers/localVolumeCleanUp.go [hunk 7]
```diff
@@ -36,57 +38,53 @@ func (cr *CleanUpOperatorReconciler) localVolumeCleanUp(ctx context.Context, nam
...
-		for _, node := range nodesList.Items {
-			command := "oc debug node/" + node.Name + " -- chroot /host rm -rf /mnt"
-			_, out, err := ExecuteCommand(command)
-			if err != nil {
-				fmt.Println("Error in removing mounted path from node: ", node.Name)
-				return err
-			}
-			fmt.Println(out)
+	for _, node := range nodesList.Items {
+		defer logFunctionDuration(cr.Log, "Mounted Path Removal", time.Now())
```
Comment (mssachan) [inline controllers/localVolumeCleanUp.go]: Dont use differ method here. Remove it

**Pull request merged**